hack/gke/xfs-format/xfs-format.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+ephemeral_mountpoint=/mnt/stateful_partition/kube-ephemeral-ssd
+raid_device=$( awk '$2=="/mnt/stateful_partition/kube-ephemeral-ssd" {print $1}' /proc/mounts )
+raid_dir=/mnt/raid-disks/disk0
+
+if [[ -z "${raid_device}" ]]; then
+  echo "Raid device not mounted"
+  exit 1
+fi
+
+if [[ $( blkid -o value -s TYPE "${raid_device}" ) == "xfs" ]]; then
+  echo "Raid device's file system is already xfs."
+  exit 0
+fi
+
+systemctl stop kube-node-configuration.service
+systemctl stop kubelet.service
+/home/kubernetes/bin/crictl stopp $( /home/kubernetes/bin/crictl pods -q )
+systemctl stop containerd.service
+
+tmp_dir=$( mktemp -d )
+cp -rf "${ephemeral_mountpoint}"/. "${tmp_dir}"/
+
+umount $( grep -e '/home/kubernetes/containerized_mounter' -e '/var/lib/containerd' -e '/home/containerd' /proc/mounts | sed -E -e 's/([^ ]+) ([^ ]+) .+/\2/' | sort -u -r ) || true
```

Comment:
would a condition on the number of returned arguments be safer then ignoring the error?

Reply:
I've changed it to a loop as some of the results wouldn't be mounted when they were called (I suppose they were binds) which would result in an error. So I'm checking for that first one by one.